### PR TITLE
refactor-2

### PR DESCRIPTION
**Pull request by sarab96**
filter session class
For interface to filter plugin
Filter to be used for generic filter use-cases even with multiple outputs

### CODE REVIEW

File: src/xma/include/lib/xma_filter_session.hpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  //user input is blank array of frames to be filled with cu output
+  //Size array is fixed; based on num_outputs variable in filter properties
+  int32_t
+  recv_frame_list(XmaFrame **frame) const;
+
+  //get_status of filter cu, kernel properties and output status
+  //user input is array of XmaParameter; User/plugin managed
+  //This optional for plugins; some plugins may not implement this functions interface
+  int32_t
+  get_status(XmaParameter *param, int32_t num_params) const;
```
Comment (stsoe) [inline src/xma/include/lib/xma_filter_session.hpp:55]: Comment says param is input, but signature appears to be output?

Comment (sarab96) [reply src/xma/include/lib/xma_filter_session.hpp:55]: Fixed comment to say it is output populated in user provided blank array

**Comment by dayeh-xilinx** [comment]
retest this please

**Comment by dayeh-xilinx** [comment]
retest this please

**Comment by xilinxgitops** [comment]
retest this please

**Pull request merged**